napari/components/_viewer/model.py

```diff
@@ -162,6 +171,27 @@ def active_layer(self, active_layer):
         self._active_layer = active_layer
         self.events.active_layer(item=self._active_layer)
 
+    @property
+    def keybindings(self):
+        return self._keybindings.copy()
+
+    @keybindings.setter
+    def keybindings(self, keybindings):
+        if keybindings == self.keybindings:
+            return
+        kb = {}
+        bind_keys(kb, keybindings.items())
+        self._keybindings = kb
+
+    bind_key = bind_key_method(keybindings='_keybindings',
+                               class_keybindings='default_keybindings')
+
+    unbind_keys = unbind_keys_method(keybindings='_keybindings',
+                                     class_keybindings='default_keybindings')
```

Comment:
Can you add some comments to these definitions? Needless to say, this is not a standard way to define methods, so some comments explaining the rationale here would be great.

Some thoughts that may or may not be relevant:
- should `keybindings=` be renamed `instance_keybindings=`?
- could this be more cleanly implemented using multiple inheritance? (ie make a `AcceptsKeybindings` class?)

Reply:
Ah yeah, I think I may go with your multiple inheritance idea.